pugaIsaias@/Entry-challenge

pugaIsaias opened the pull request:
- First time using typescript, and first time doing a pull request on Github.
- Tried to use express, just to find out its a listener. And not a package to do commands.
- Used fetch to do both the POST and GET.
- Read the code on Index.ts to find out the JSON path. And retrieve the claveDeAcceso with GET.
- Package.json may be be able to reduce since I don't use most of the imports.

pugaIsaias committed:
Primer dia de codigo

- se creo client.jts
- utiliza express para realizar los envios al servidor. Ya que lo utiliza index.ts
- solo para terminar , leer la documentacion y saber routing y express
- aun no funciona ni post,  ni get

pugaIsaias committed:
POST & GET done! 

- Used async & await fetch. Cause easy paramethers, and call.
- Learned that catch was needed on the async function call, not declaration.

and here is a resource where I learn it https://www.youtube.com/watch?v=tc8DU14qX6I

pugaIsaias committed:
POST & GET done!

- repository didn't include client.ts, include it manually.
- package-lock & package.json update

pugaIsaias committed:
Merge branch 'master' of https://github.com/pugaIsaias/entry-challenge-v1

pugaIsaias committed:
Got claveDeAcceso

Changed the GET response JSON leafs

MConcoba commented:
@pugaIsaias Buen comienzo,
Soluciona los siguientes comments

MConcoba commented on src/client.js:
```diff
@@ -0,0 +1,33 @@
+});
+app.get(url + '/' + data_json_1["default"].contactInfo.emailAddress, function (req, res) {
+    res.send('Hello World!');
+});
```
En las instrucciones pide crear un archivo `./src/client.ts` el cual ya lo tienes,
Este archivo no es necesario, recuerda que estamos trabajando con `TypeScript`

pugaIsaias replied:
Muy bien ya elimine los archivo `.js` que se crean cuando le doy `tsc client.ts ` en la consola.

pugaIsaias replied:
Ya elimine todos los `.js `que se crearon cuando corria `tsc client.ts` en mi terminal.

MConcoba commented on src/data.json:
```diff
@@ -0,0 +1,30 @@
+{
+    "contactInfo": {
+        "fullName": "Isaias Samuel Puga Solares",
+        "emailAddress": "puga.isaias@gmail.com"
+    },
+    "github": {
+        "profileURL": "https://github.com/pugaIsaias",
+        "username": "pugaIsaias"
+    },
+    "credentials": {
+        "password": "4MXw5kW9"
+    },
+    "personalInfo": {
+        "questions": [
+            {
+                "question": "If I was a Sr. Programmer, I would like to build:",
+                "answer": "Respuesta: Develop games, and good experiences"
+            },
+            {
+                "question":
+                "Por favor indica el URL que me lleva a la línea de código de la definición de React.useEffect",
+                "answer": "Respuesta: ..."
+            },
+            {
+                "question": "code is poetry, because:",
+                "answer": "Respuesta: It has free form, can be fun, or tragic. And with words you build something that can be enjoyed by yourself, or others."
+            }
+        ]
+    }
+}
```
Modifica este archivo a `data.ts`

netpoe replied:
@MConcoba hay que ser más claros con esta instrucción: 

> Cambia el nombre de este archivo de `data.json` a `data.ts`

MConcoba replied:
@netpoe ok, entendido

pugaIsaias replied:
LIsto

MConcoba commented on src/data.json:
```diff
@@ -0,0 +1,30 @@
+    "github": {
+        "profileURL": "https://github.com/pugaIsaias",
+        "username": "pugaIsaias"
+    },
+    "credentials": {
+        "password": "4MXw5kW9"
+    },
```
Ingresar una contraseña directamente en el código, es una mala practica de seguridad, hay que buscar una manera para no introducirla de manera directa!

pugaIsaias replied:
Listo, y utilize el `.env` para manejar la contraseña.

MConcoba commented on src/data.json:
```diff
@@ -0,0 +1,30 @@
+                "question": "If I was a Sr. Programmer, I would like to build:",
+                "answer": "Respuesta: Develop games, and good experiences"
+            },
+            {
+                "question":
+                "Por favor indica el URL que me lleva a la línea de código de la definición de React.useEffect",
+                "answer": "Respuesta: ..."
+            },
```
Debes de completar todas las preguntas, te falta esta....

pugaIsaias replied:
Listo

MConcoba commented on src/client.ts:
```diff
@@ -0,0 +1,57 @@
+//how to import JSON
+const data = require("./data.json");
+const email = data.contactInfo.emailAddress;
+console.log(email); // output 'testing'
```
El uso de `variables globales `en JS, no es una buena practica lo mejor es declararlas en `el scope de la función donde se van a utilizar` o en este caso como es una `variable de entorno` puedes simplemente usarla dentro de tu función donde la necesites.

pugaIsaias replied:
`const url `se movio dentro de cada una de las funciones.

Pregunta: los `import` tambien se hacen a nivel de funcion o estos estan bien a nivel global?

MConcoba replied:
No, los `import`, estan bien a nivel global

MConcoba commented on src/client.ts:
```diff
@@ -0,0 +1,57 @@
+
+postData()
+.catch((err) => {console.log('ERROR: ', err.message)}); // need catch or it won't work
+
+getData()
+.catch((err) => {console.log('ERROR: ', err.message)}); // need catch or it won't work
```
Algo que no es obligatorio, pero da buena estética al código es agregar una linea en blanco al final de los archivos de código, te recomiendo que lo hagas!

pugaIsaias replied:
OK lo voy a tomar en cuenta. 
Pero creo GitHub elimino mi linea en blanco en este commit.

pugaIsaias committed:
Soluciono comentarios

- Se elimina archivos .js
- data.json ahora es data.ts
- se utiliza .env para manejar el password
- se agrega solucion a pregunta de linea de codigo de la definicion de React.useEffect
- Se agrego linea vacia al final de los archivo .ts

pugaIsaias commented:
- Se elimina archivos `.js`
- `data.json `ahora es `data.ts`
- se utiliza `.env` para manejar el password
- se agrega solucion a pregunta de linea de codigo de la definicion de React.useEffect
- Se agrego linea vacia al final de los archivo `.ts`

MConcoba commented on src/client.ts:
```diff
@@ -37,12 +28,25 @@ async function postData(){
         let json = await response.json();
-        console.log(json.bienvenido.claveDeAcceso);
         console.log("Did a get.");
+        console.log(json.bienvenido.claveDeAcceso);
```
Hay que procurar no dejar `console.logs.`
Vamos a aprender a utilizar herramientas de debugging para no utilizar `console.logs`

pugaIsaias replied:
Ok, pero por el momento. Solo elimino los `console.log`?

MConcoba commented on src/client.ts:
```diff
@@ -37,12 +28,25 @@ async function postData(){
         let json = await response.json();
-        console.log(json.bienvenido.claveDeAcceso);
         console.log("Did a get.");
+        console.log(json.bienvenido.claveDeAcceso);
```
Hay que procurar no dejar `console.logs.`
Vamos a aprender a utilizar herramientas de debugging para no utilizar `console.logs`

MConcoba commented on src/client.ts:
```diff
@@ -1,30 +1,21 @@
-        "x-password": data.credentials.password
-    }    
-}
+    //optional paramethers
```
Procura no utilizar `comentarios`, ya que el codigo se tiene que explicar por si mismo

pugaIsaias replied:
Y elimino `comentarios`

pugaIsaias committed:
Soluciono comentarios 2

- Elimino console.log ya que depurara de otra forma.

- Se elimina los comentarios.

-Adicionalmente elimine alguna Dependencias de cosas que ya no utilize (express, body-parser, axios) del package.json. Y aun regresa la respuesta.

- Y intento agregar 2 espacios al final del codigo. A ver como lo recibe GitHub

netpoe commented on src/client.ts:
```diff
@@ -0,0 +1,56 @@
+
+    const email = data.contactInfo.emailAddress;
+    let response = await fetch(url+email,getParamethers);
+    if (response.ok) { 
```
Estos bloques de `if, else` no son necesarios. Para eso existe el `try/catch` 
Remuévelos por favor.

pugaIsaias replied:
@netpoe  Muy bien ,muchas gracias. Con eso ya puedo quitar el `.catch `que tenia que utilizar al llamar las funciones.

netpoe commented on src/client.ts:
```diff
@@ -0,0 +1,56 @@
+
+    let response = await fetch(url,postParamethers);
+    let result = await response.json();
+    if (response.ok) {
```
Igualmente, estos bloques no son necesarios.

netpoe commented on src/data.ts:
```diff
@@ -0,0 +1,35 @@
+    }
+}
+
+export { data };
```
```suggestion
export default data;
```

pugaIsaias replied:
Elimine por completo esta linea. Ya que la exportaba directamente en la siguiente corrección.

netpoe commented on src/data.ts:
```diff
@@ -0,0 +1,35 @@
+require('dotenv').config();
+
+const data ={
```
```suggestion
export const data = {
```

pugaIsaias replied:
Muy bien ya realize este cambio. Y elimine el export que realizaba al final para evitar redundancia.

netpoe commented on src/data.ts:
```diff
@@ -0,0 +1,35 @@
+        "username": "pugaIsaias"
+    },
+    "credentials": {
+        "password": process.env.API_KEY
```
Este sería un nombre más adecuado: 

```suggestion
        "password": process.env.PASSWORD
```

pugaIsaias replied:
Muy bien ya corregí el nombre de la variable.

netpoe reviewed (changes requested):
@pugaIsaias resuelve los comentarios. 

También, `package-lock.json` no es necesario, para esto estamos usando `yarn` y `yarn` crea su propio `yarn.lock`. 

Igualmente, por favor instala https://github.com/prettier/prettier-vscode en tu editor y agrega estas líneas en tu `settings.json`: 

```
"editor.codeActionsOnSave": {
    "source.organizeImports": true
  },
```

pugaIsaias commented:
- Se elimina `console.log`. Ya que se depurara de otra forma.

- Se elimina los comentarios. Ya que se debe a dar a enteder si ellos.

- Adicionalmente elimine algunas Dependencias de cosas que ya no utilize (express, body-parser, axios) del `package.json. ` Sin consecuencias al resultado.

- Y intento agregar 2 espacios al final del codigo. A ver como lo recibe GitHub.

MConcoba commented on src/client.ts:
```diff
@@ -19,41 +17,40 @@ async function postData(){
+        
     } else {
-        console.log("HTTP-Error: "+ response.status);
+        
```
Los bloque de `if, else` que tienes, no son necesarios, podrias eliminarlos

pugaIsaias replied:
LIsto

MConcoba commented on package-lock.json:
```diff
@@ -0,0 +1,534 @@
+{
```
este file no debería estar en este PR. No hay necesidad de modificar el package.json. Además estamos usando yarn y éste genera yarn.lock.

pugaIsaias replied:
LIsto eliminado.

MConcoba commented:
@pugaIsaias  puedes instalar la extensión `prettier `en tu editor.

También puedes agregar esto a tus settings.json:

```
"editor.codeActionsOnSave": {
    "source.organizeImports": true
  },
```

pugaIsaias committed:
Soluciono comentarios 3

Se instalo prettier y se modifco el setting.json

En data.ts
- cambio el export del final hacia su declaracion.

En package-lock.json
- Se elimina ya que no es necesario.

En client.ts
- Se elimina los if/else y se sustituye en try/catch.
- Se elimina el catch que se hacia fuera de funcion. Ya que ahora se hace adrentro de la funcion y no da errores.

pugaIsaias reviewed (commented):
Cambio hechos. Favor revisar cualquier otro cambio que se necesite.

netpoe reviewed (changes requested):
@pugaIsaias muy cerca. 
por favor instala `prettier` y al hacer save de tus files deberías ver cómo se alinea tu código.

pugaIsaias committed:
Se aplica prettier

netpoe reviewed (commented):
👍 

@pugaIsaias buen trabajo. 

Revisa tu correo con las invitaciones a Slack y Github.
